demos/interactive.py

```diff
@@ -48,6 +50,11 @@ def posix_shell(chan):
         chan.settimeout(0.0)
 
         while True:
+            # Resize terminal size every loop to make change timely
+            s = struct.pack("HHHH", 0, 0, 0, 0)
+            t = fcntl.ioctl(sys.stdout.fileno(), termios.TIOCGWINSZ, s)
+            winsize = struct.unpack("hhhh", t)
+            chan.resize_pty(width=winsize[1], height=winsize[0])
```

Comment:
This is just a demo but ... this is a lot of redundant traffic to send over the channel. This should only be sent if width or height changes. But also, it's a lot of ioctl() syscalls to check every loop too, I think you can add a signal handler for SIGWINCH to get notifications that terminal size changed.